internal/authtoken/query.go

```diff
@@ -0,0 +1,178 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package authtoken
+
+const (
+	estimateCountAuthTokens = `
+select reltuples::bigint as estimate from pg_class where oid in ('auth_token'::regclass)
+`
+	listAuthTokensTemplate = `
+with auth_tokens as (
+     select public_id,
+            auth_account_id,
+            create_time,
+            update_time,
+            approximate_last_access_time,
+            expiration_time,
+            status
+       from auth_token
+   order by create_time desc, public_id asc
+      limit %d
+),
+auth_accounts as (
+    select public_id,
+           auth_method_id,
+           scope_id,
+           iam_user_id,
+           iam_user_scope_id
+      from auth_account
+     where %s
```

Comment:
Nice, this is better, but I think we can do one better with 
```suggestion
     where %s
       and public_id in (select auth_account_id from auth_tokens)
```
What do you think? That will only select the auth accounts that we actually need.